changelog.d/9659.misc

```diff
@@ -0,0 +1 @@
+Add and fix more bugbear lint violations.
```

Comment:
To match #9499:

```suggestion
Introduce bugbear to the test suite and fix some of it's lint violations.
```

Reply:
I'll duplicate this on every subsequent PR, then :+1:

Reply:
They get merged automatically by towncrier so this makes it a bit easier.